src/Tasks/Microsoft.NET.Build.Tasks/EmbedAppNameInHost.cs

```diff
@@ -59,7 +60,24 @@ protected override void ExecuteCore()
             {
                 Directory.CreateDirectory(destinationDirectory);
             }
-            File.WriteAllBytes(ModifiedAppHostPath, array);
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I would keep the code consistent between platforms too. I don't know what you mean by "no overhead", though. We're copying the whole file only to replace its contents completely. We can fix this by finding the index of the marker, seeking to the right position and overwriting it.

On a related note, I pointed out in the original code review that reading the whole file in to a managed byte array was wasteful, but that was never addressed.